Client/Frontend/ContentBlocker/ContentBlockerHelper.swift

```diff
@@ -423,14 +326,73 @@ extension ContentBlockerHelper {
         }
     }
 
-    func whitelistJSON() -> String {
+}
+
+// MARK: Whitelisting support
+@available(iOS 11.0, *)
+extension ContentBlockerHelper {
+
+    static func whitelistFileURL() -> URL? {
+        guard let dir = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first else {
+            Sentry.shared.send(message: "Failed to get doc dir for whitelist file.")
+            return nil
+        }
+        return dir.appendingPathComponent("whitelist")
+    }
+
+    fileprivate func whitelistJSON() -> String {
         if ContentBlockerHelper.whitelistedDomains.isEmpty {
             return ""
         }
         // Note that * is added to the front of domains, so foo.com becomes *foo.com
         let list = "'*" + ContentBlockerHelper.whitelistedDomains.joined(separator: "','*") + "'"
-        return ", {'action': { 'type': 'ignore-previous-rules' }, 'trigger': { 'url-filter': '.*', 'unless-domain': [\(list)] }".replacingOccurrences(of: "'", with: "\"")
+        return ", {'action': { 'type': 'ignore-previous-rules' }, 'trigger': { 'url-filter': '.*', 'unless-domain': [\(list)] }".escapeJSON()
+    }
+
+    func whitelist(enable: Bool, forDomain domain: String, completion: (() -> Void)? = nil) {
+        if enable {
+            ContentBlockerHelper.whitelistedDomains.append(domain)
+        } else {
+            ContentBlockerHelper.whitelistedDomains = ContentBlockerHelper.whitelistedDomains.filter { $0 != domain }
+        }
+
+        BlockListChecker.shared.whitelistedDomains = ContentBlockerHelper.whitelistedDomains
```

Comment:
Both classes keep a list of whitelistedDomains? Should only be one list no?

Reply:
Oops, this was in the original code, looks like it is due to the separation of Focus code vs 'our code'. This seemed like a useful distinction when the Focus stats code was added, but I don't think it is anymore -we ended up with various modifications to integrate with our code.